### PR TITLE
feat(deepcast): more input and output options

### PR DESCRIPTION
**Pull request by erics118**
## Description

- add ability to choose where to get text from, as a preferred source, eg get from selected text first, but then fallback to clipboard, or the other way around.
- add ability to paste text when done translating (best when preferred source is selected text)

## Screencast

todo

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/deepcast/tsconfig.json [hunk 1]
```diff
@@ -1,16 +1,19 @@
...
     "strict": true,
     "isolatedModules": true,
     "esModuleInterop": true,
     "skipLibCheck": true,
     "forceConsistentCasingInFileNames": true,
-    "jsx": "react-jsx"
+    "jsx": "react-jsx",
+    "resolveJsonModule": true,
+    "typeRoots": ["."],
+    "types": ["raycast-env"]
```
Comment (erics118) [inline extensions/deepcast/tsconfig.json:17]: This was added to take advantage of the autogenerated types in `raycast-env.d.ts`. `ray dev` uses them just fine, but the editor needs to know that it exists by adding these lines.

File: extensions/deepcast/src/utils.ts [hunk 2]
```diff
@@ -6,19 +6,11 @@ import {
...
   LaunchType,
+  closeMainWindow,
 } from "@raycast/api";
 import got from "got";
 import { StatusCodes, getReasonPhrase } from "http-status-codes";
 
-interface Preferences {
-  key: string;
-  onTranslateAction: "clipboard" | "view";
-}
```
Comment (erics118) [inline extensions/deepcast/src/utils.ts:16]: The addition to `tsconfig.json` makes this interfaace unnecessary.

File: extensions/deepcast/package.json [hunk 3]
```diff
@@ -266,25 +267,49 @@
...
+          "title": "Selected",
+          "value": "selected"
+        },
+        {
+          "title": "Clipboard",
+          "value": "clipboard"
+        }
+      ],
+      "default": "selected",
+      "required": false
```
Comment (erics118) [inline extensions/deepcast/package.json:293]: This was copied over from my `change-case` extension

**Comment by erics118** [comment]
I was thinking about removing all the individual commands and instead using quicklinks. This would make the codebase a lot easier, but a migration script probably would need to be written to automatically convert existing enabled commands to quicklinks, and we need a good UX for making these quicklinks

**Comment by erics118** [comment]
CI fails, but idk what to say except that it works on my computer 😅


CleanShot 2023-10-04 at 18 09 56

File: extensions/deepcast/tsconfig.json (hunk 1 above)
Comment (pernielsentikaer) [reply extensions/deepcast/tsconfig.json:17]: I'm not sure I understand why you need this.

If something is not found, you should be able to run `> ESLint: Restart ESLint Server` in code.

**Comment by pernielsentikaer** [comment]
@erics118 do you mind adding a CHANGELOG entry too?

@mooxl do you mind checking this?

**Comment by pernielsentikaer** [comment]
@mooxl could you check this?

**Comment by mooxl** [comment]
LGTM!

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

**Pull request merged**